source/GlassView.Export/Serialization/BenchmarkSerializer.cs

```diff
@@ -1,22 +1,21 @@
 using System.Text.Json;
 using GlassView.Core.Models;
-using GlassView.Export.Models;
 
 namespace GlassView.Export.Serialization;
 
-internal sealed class BenchmarkSerializer : System.Text.Json.Serialization.JsonConverter<Benchmark>
+internal sealed class BenchmarkSerializer : System.Text.Json.Serialization.JsonConverter<BenchmarkSummary>
 {
-    public override void Write(Utf8JsonWriter writer, Benchmark value, JsonSerializerOptions options)
+    public override void Write(Utf8JsonWriter writer, BenchmarkSummary value, JsonSerializerOptions options)
     {
         writer.WriteStartObject();
         WriteBenchmark(new JsonWriter(writer), value);
         writer.WritePropertyName(nameof(value.Environment));
```

Comment:
I don't understand why we need to write the Environment separately. It seems to be still part of the same model, `Benchmark value`. Could you please explain :) ?

Reply:
This is a `System.Text.Json` quirk.

- See [here](https://stackoverflow.com/questions/74615202/system-text-json-how-to-serialise-an-ienumerable-like-a-regular-class) on stackoverflow
- that links to the relevant (closed) [issue](https://github.com/dotnet/runtime/issues/1808) on the dotnet repo.
- which will be resolved by this (open) [issue](https://github.com/dotnet/runtime/issues/63791).

For now, we're stuck with either re-modelling our models (easy, but not "clean" IMHO) or writing a custom converter (hard, but "clean" IMHO).